packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1224,6 +1227,9 @@ export function getClientNamespace(
 }
 
 function getNamespaceFullNameWithOverride(context: TCGCContext, namespace: Namespace): string {
+  if (context.namespace) {
+    return context.namespace;
+  }
   const segments = [];
   let current: Namespace | undefined = namespace;
   while (current && current.name !== "") {
```

Comment:
I assume the `@clientNamespace` (if specified on TypeSpec Namespace or Model) is having higher priority than `namespace` in emitter option.

@srnagar for confirmation.